### PR TITLE
Colon bindings

### PR DESCRIPTION
**Pull request by justinbmeyer**
For #229 

- [x] - Bring back other tests.
- [x] - Tests for `<div on:born:by:person="animate('slowly')"/>`
- [x] - Make it able to bind directly to the element if there is no view model.
- [x] - Write tests for `getBindingInfo`.
- [x] - What should the behavior now be of `on:someEvent` `on:some-event`.
- [x] - What is the behavior of `{fooBar}`, `{foo-bar}`, `(someEvent)`, `(some-event)`.
- [x] - How should we best encode / decode to the DOM.
       - In can-view-parser if someone has `(foo bar)="zed"` -> `(foo\\sbar)`
       - `on:someEvent` -> `on:some\\cevent`
       - Why in can-view-parser: Because we allow `<div {{#if foo}}on:someEvent="doSomething()"{{/if}}>`

- [ ] - Other binding specifiers:
       - `<input on:enter:value:to="scopeValue"/>`
       - Specify winner in initialization or stickiness  with `<location-feed location:bind="userLocation">`
       -  stickiness with something like `<input value:bind="age"/>`
       - Set prop - `<input prop:value:bind="age"/>`
       - Custom events and props:
          ```
          <can-import viewModel:to="drag" from="can-dom/drag/event"/>
          <div on:[drag]="doSomething()"/>
          ```

### CODE REVIEW

**Comment by phillipskevin** [comment]
@justinbmeyer this is ready for review if you can take a look. (can't add you as a reviewer since you opened the PR initially).

File: can-stache-bindings.js [hunk 1]
```diff
@@ -265,13 +269,17 @@ var behaviors = {
 			if(domData.get.call(el, "preventDataBindings")) {
 				return;
 			}
-			var viewModel = canViewModel(el),
+			var viewModel,
+				getViewModel = function() {
+					return viewModel || (viewModel = canViewModel(el));
+				},
```
Comment (justinbmeyer) [inline can-stache-bindings.js]: why is this needed?

File: can-stache-bindings.js [hunk 2]
```diff
@@ -351,27 +355,53 @@ var behaviors = {
...
+				// the name of the event we are binding
+				event,
+				// if we are binding on the element or the VM
+				bindingContext;
+
+			// legacy binding
+			if(attributeName.indexOf('can-') === 0) {
+				event = attributeName.substr("can-".length);
+				bindingContext = el;
+			} else if(attributeName.indexOf("on:") === 0 ) {
```
Comment (justinbmeyer) [inline can-stache-bindings.js]: I think there wouldn't be much benefit, but we might want to cache `":on"` in the outer scope so we can do:

`.indexOf(onMatchStr)` and `onMatchStr.length`

File: package.json [hunk 3]
```diff
@@ -31,28 +31,29 @@
     "main": "can-stache-bindings"
   },
   "dependencies": {
+    "can-attribute-encoder": "^0.3.0",
```
Comment (justinbmeyer) [inline package.json:34]: we should hard-code core deps that are not `>1.0` (we don't do this everywhere (can-stache-key), but we should)

File: can-stache-bindings.js (hunk 1 above)
Comment (phillipskevin) [reply can-stache-bindings.js]: Because otherwise a viewModel will always be set up on the element and we will always use `vm` binding instead of element binding.

File: package.json (hunk 3 above)
Comment (phillipskevin) [reply package.json:34]: I released this as > `0.1.0` so that we could publish patches for docs and things.

Comment (justinbmeyer) [reply package.json:34]: what do you mean?  It should be hard-coded because the API might change.  Everything included by core shouldn't break if it gets updated.

Comment (phillipskevin) [reply package.json:34]: If the API changes, we should publish `0.4.0`. In that case, it wouldn't get updated here.

Not hard-coding it allows us to publish bug fixes and doc updates.
